plugins/modules/dedicated_server_boot.py

```diff
@@ -33,74 +34,103 @@
         description:
             - When you want to force a dedicated server reboot
 
-'''
+"""
 
-EXAMPLES = r'''
+EXAMPLES = r"""
 - name: Change the bootid of a dedicated server to rescue
   synthesio.ovh.dedicated_server_boot:
     service_name: "{{ service_name }}"
     boot: "rescue"
     force_reboot: "true"
   delegate_to: localhost
-'''
+"""
+
+RETURN = """ # """
+
+from ansible_collections.synthesio.ovh.plugins.module_utils.ovh import (
+    OVH,
+    ovh_argument_spec,
+)
 
-RETURN = ''' # '''
 
-from ansible_collections.synthesio.ovh.plugins.module_utils.ovh import OVH, ovh_argument_spec
+def build_boot_list(service_name: str, client: OVH) -> dict:
+    """Build a list of available boot option available.
+    Get available netboots, for eatch one fetch its ID.
+    """
+    netboot = dict()
+    boot_ids = client.wrap_call("GET", f"/dedicated/server/{service_name}/boot")
+    for boot_id in boot_ids:
+        boot_infos = client.wrap_call(
+            "GET", f"/dedicated/server/{service_name}/boot/{boot_id}"
+        )
+        if boot_infos["kernel"] == "hd":
+            # The kernel parameter is "hd" in the API
+            # but we use "harddisk", which is the bootType parameter
```

Comment:
maybe by using `hd` in the call, we don't have to "translate" `harddisk` to `hd` inside the module :thinking:

Reply:
As discussed IRL, Removing this to ease maintenance.  
Every netboot uses the `kernel` parameter to retieve the corresponding bootid.